ext/bcmath/libbcmath/src/bcmath.h

```diff
@@ -79,6 +79,53 @@ typedef struct bc_struct {
 #define LONG_MAX 0x7ffffff
 #endif
 
+#if defined(_MSC_VER)
+#  include <stdlib.h>
+#  define BSWAP32(u) _byteswap_ulong(u)
+#  define BSWAP64(u) _byteswap_uint64(u)
+#else
+#  ifdef __has_builtin
+#    if __has_builtin(__builtin_bswap32)
+#      define BSWAP32(u) __builtin_bswap32(u)
+#    endif // __has_builtin(__builtin_bswap32)
+#    if __has_builtin(__builtin_bswap64)
+#      define BSWAP64(u) __builtin_bswap64(u)
+#    endif // __has_builtin(__builtin_bswap64)
+#  elif defined(__GNUC__) && ( \
+                    __GNUC__ > 4 || ( \
+                      __GNUC__ == 4 && ( \
+                        __GNUC_MINOR__ >= 3 \
+                      ) \
+                    ) \
+                  )
+#    define BSWAP32(u) __builtin_bswap32(u)
+#    define BSWAP64(u) __builtin_bswap64(u)
+#  endif // __has_builtin
```

Comment:
Is this really worthwhile? GCC 4.4.3 is old (January 2010, over 14y ago), and convoluting the code like this for an optimization doesn't look amazing to me.

Reply:
I didn't check it properly because I copied it directly from ext/hash. thank you.